### PR TITLE
[WIP] Abstract HWOBJ 6

### PR DESCRIPTION
**Pull request by rhfogh**
Proposal for AbstractActuator (Abstract hardware object) to incorporate discussion at Hamburg meeting

  - There are only two abstract methods, get_value and _set_value

  - State changing, limits setting and limits handling must also be done in subclasses

  - The  value is stored inside this class *only* for mock objects - I am assuming that for real hardware the value will be got from / set on the hardware control system.

Example include a general mock actuator, a specific implementation of the latter MockTransmission) and an attempt to use the new abstract class to implement an ExporterMotor. NB, the latter could need very thorough code review, in the (not unlikely) case I have misunderstood something

### CODE REVIEW

**Comment by rhfogh** [comment]
As of 20190705 this PR does not fit in with the existing superclass structure for HardwareObjects, and so will niot work. I shall made a working version ASAP. Meanwhile, please be patient.

**Comment by rhfogh** [comment]
The PR is now ready for looking at

I have 

- Added the correct superclasses for the examples

- renamed CommandContainer.get_value and .set_value to .get_channel_value and .set_channel_value, so the names do not clash with the new get_value and set_value

- refactored HardwareObjectNode and HardwareObject in preparation for possibly later replacing HardwareObjectNode with the new ConfiguredObject class (otehr PR)., We have not decided whether to do that yet, but if we are going to rewrite a lot of HardwareObeject classes, we might as well make the few innocuouc changes that would get us ready. If we do decide to do it, we should eventually remove HardwareObjectNode as a superclass of HardwareObject

File: BaseHardwareObjects.py [hunk 1]
```diff
@@ -264,22 +264,24 @@ def getProperty(self, name, default_value=None):
...
-           The easiest solution is to call update_values method directly
-           after getHardwareObject and connect.
-
-           Normaly this method would emit all values
-        """
-        return
-
-    def clear_gevent(self):
-        pass
+    # Moved to HarwareObject, DeviceContainerNode for refactoring
```
Comment (IvarsKarpics) [inline BaseHardwareObjects.py]: Maybe better remove all commented lines?

File: BaseHardwareObjects.py [hunk 2]
```diff
@@ -498,7 +523,30 @@ def getDeviceByRole(self, role):
...
+
+    def clear_gevent(self):
+        """Clear gevent tasks
+
+        Returns:
+
+        """
+        pass
+
+    def update_values(self):
```
Comment (IvarsKarpics) [inline BaseHardwareObjects.py]: At some point we have to rename it to "reemit_values" or something similar.

File: CommandContainer.py [hunk 3]
```diff
@@ -352,19 +352,19 @@ def add_channel(self, attributesDict, channel, addNow=True):
...
         self.__channels[channel_name].setValue(value)
 
-    def get_value(self, channel_name):
+    def get_channel_value(self, channel_name):
         return self.__channels[channel_name].getValue()
 
     def setValue(self, channelName, value):
         warn("setValue is deprecated. Use set_value instead", DeprecationWarning)
-        self.set_value(channelName, value)
+        self.set_channel_value(channelName, value)
```
Comment (IvarsKarpics) [inline CommandContainer.py]: Why not remove setValue and getValue at all?

File: HardwareObjects/ExporterMotorExample.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  <exporter_address>wid30bmd2s:9001</exporter_address>
+  <motor_name>AlignmentY</motor_name>
+  <GUIstep>1.0</GUIstep>
+  <unit>-1e-3</unit>
+  <resolution>1e-2</resolution>
+</device>
+"""
+
+
+class ExporterMotorExample(AbstractActuator2.StandardHardwareObject):
```
Comment (IvarsKarpics) [inline HardwareObjects/ExporterMotorExample.py]: Why we need an Example hwobj? Is this a mockup? Why not change ExporterMotor directly?

File: HardwareObjects/abstract/AbstractActuator2.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+from somewhere import GeneralState
+
+__credits__ = [" Copyright © 2019 by MXCuBE collaboration. All rights reserved"]
+__license__ = "LGPLv3+"
+__category__ = "General"
+__author__ = "rhfogh"
+__date__ = "20190607"
+
+
+class AbstractActuator(HardwareObject, HardwareObjectNode):
```
Comment (IvarsKarpics) [inline HardwareObjects/abstract/AbstractActuator2.py]: Class name is AbstractActuator and filename is AbstractActuator2. Bug or feature? I guess it should not be 2?

File: HardwareObjects/mockup/AbstractMockActuator.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+from HardwareRepository.HardwareObjects.abstract import AbstractActuator2
+
+__credits__ = [" Copyright © 2019 - by MXCuBE collaboration. All rights reserved"]
+__license__ = "LGPLv3+"
+__category__ = "General"
+__author__ = "rhfogh"
+__date__ = "17/05/2019"
+
+
+class AbstractMockActuator(AbstractActuator2.AbstractActuator):
```
Comment (IvarsKarpics) [inline HardwareObjects/mockup/AbstractMockActuator.py]: AbstractMockActuator? Hmm, Abstract or Mockup, but not both.

File: HardwareObjects/mockup/TransmissionMockup.py [hunk 7]
```diff
@@ -1,46 +1,49 @@
...
+from HardwareRepository.HardwareObjects.mockup import AbstractMockActuator
+
+__credits__ = [" Copyright © 2019 -  by MXCuBE collaboration. All rights reserved"]
+__license__ = "LGPLv3+"
+__category__ = "General"
+__author__ = "rhfogh"
+__date__ = "17/05/2019"
+
+
+class TransmissionMockup(AbstractMockActuator.AbstractMockActuator):
```
Comment (IvarsKarpics) [inline HardwareObjects/mockup/TransmissionMockup.py]: AbstractActuator.AbstractActuator

**Comment by rhfogh** [comment]
Superseded by #420

**Pull request closed without merging**